be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala

```diff
@@ -47,7 +52,64 @@ case object RollCallHandler extends MessageHandler {
   }
 
   def handleCloseRollCall(rpcMessage: JsonRpcRequest): GraphMessage = {
-    val ask: Future[GraphMessage] = dbAskWritePropagate(rpcMessage)
-    Await.result(ask, duration)
+    rpcMessage.getParamsMessage match {
+      case Some(message: Message) =>
+        val data: CloseRollCall = message.decodedData.get.asInstanceOf[CloseRollCall]
+        val unknownAnswer: String = "Database actor returned an unknown answer"
+
+        //Creates a channel for each attendee (of name /root/lao_id/social/PublicKeyAttendee), returns a GraphMessage
+        def createAttendeeChannels(attendees: List[PublicKey], rpcMessage: JsonRpcRequest): GraphMessage = {
+          attendees match {
```

Comment:
This function looks strange/messy/out of place. Is there a better way to do it?

Reply:
I've been thinking about this and will talk about it during our meeting, the problem being that each channel needs to be created separately and we should take into account the failure directly, in order to prevent problems later on. The "easiest" way I found for now was therefore to directly return a GraphMessage, however, the function is still a bit messy I agree with you